### PR TITLE
Use the fparser package and remove f2py

### PR DESCRIPTION
**Pull request by arporter**
Pull request for work done under #8.

### CODE REVIEW

**Comment by arporter** [comment]
In preparing this pull request I've realised that our documentation is now out-of-date since it talks about svn rather than git. I've created #9 to fix this.

File: doc/getting_going.rst [hunk 1]
```diff
@@ -21,48 +21,48 @@ PSyclone directory itself) will be refered to as <PSYCLONEHOME>
...
-PYTHONPATH variable to include this directory.
+The fparser package (https://github.com/stfc/fparser) is a Fortran
+parser originally developed as a part of the f2py project. PSyclone
+requires version >= 0.0.2. It is available from the Python Package
+Index and thus may be installed using pip
+(https://packaging.python.org/installing/#requirements-for-installing-packages):
 ::
 
-    > export PYTHONPATH=<PSYCLONEHOME>/f2py_93:${PYTHONPATH}
+    > pip install --user fparser
```
Comment (rupertford) [inline doc/getting_going.rst]: Show how to do a system wide install if they have appropriate admin right, then say, they can do a local install if they don't have permission and show the user option. You could also show how to uninstall as well.

File: doc/getting_going.rst [hunk 2]
```diff
@@ -21,48 +21,48 @@ PSyclone directory itself) will be refered to as <PSYCLONEHOME>
...
 PSyclone requires pyparsing, a library designed to allow parsers to be be
 built in Python. PSyclone uses pyparsing to parse fortran regular
-expressions as f2py does not fully parse these, (see
+expressions as fparser does not fully parse these, (see
 http://pyparsing.wikispaces.com for more information).
 
-PSyclone has been tested with pyparsing version 1.5.2 which is a
-relatively old version but is currently the version available in the
-Ubuntu software center.
+PSyclone has been tested with pyparsing versions 1.5.2 and 2.0.1.
```
Comment (rupertford) [inline doc/getting_going.rst]: After adding this line, a later line no longer makes sense so should be deleted ...
"Versions higher than 1.5.2 should work but have not been tested"

File: doc/getting_going.rst [hunk 3]
```diff
@@ -71,8 +71,9 @@ higher than 1.5.2 should work but have not been tested.
 
 If pyparsing is not installed on your system you can install it from
 within Ubuntu using the software center (search for the
-"python-pyparsing" module in the software center and install). If you
-do not run Ubuntu you could follow the instructions here
+"python-pyparsing" module in the software center and install). It is
```
Comment (rupertford) [inline doc/getting_going.rst]: remove the Ubuntu-specific install info and show how to install using pip. Again show system wide if admin rights, then user install only and finally show how to uninstall.

File: doc/system_specific_setup.rst [hunk 4]
```diff
@@ -50,6 +56,41 @@ pyparsing is required by PSyclone
...
+
+Install fparser
+^^^^^^^^^^^^^^^
+
+fparser is also required by PSyclone but is not available from the
+Ubuntu software centre. It can instead be installed from the
+Python Package Index using pip:
+::
+
+    > pip install --user fparser
```
Comment (rupertford) [inline doc/system_specific_setup.rst]: Show how to install system wide first saying if you have appropriate permissions, then user then how to uninstall.

File: doc/system_specific_setup.rst [hunk 5]
```diff
@@ -137,4 +181,8 @@ You can now install the extensions
     > sudo pip install pytest-flakes
     > sudo pip install pytest-pep257
 
+If you don't have root access then you can specify the ``--user`` argument to
+install packages in a user-local directory -- see the instructions on
+:ref:`install_fparser` above.
```
Comment (rupertford) [inline doc/system_specific_setup.rst]: for consistency also show how to uninstall. I know this should have been there before.

File: src/f2pygen.py [hunk 6]
```diff
@@ -12,7 +12,7 @@
 
 from fparser.statements import Comment
 from fparser.readfortran import FortranStringReader
-from fparser.block_statements import Select
+from fparser.block_statements import SelectCase, SelectType
```
Comment (rupertford) [inline src/f2pygen.py]: f2pygen now has it's own repo and is available via pip so this is a temporary fix until we remove the embedded f2pygen.py code and use the remote one. An issue explaining the problem with Select for the new version of fparser has been created in the fgenerator project so all is well.

File: src/tests/f2pygen_test.py [hunk 7]
```diff
@@ -1077,8 +1077,8 @@ def test_selectiongen_addcase():
     sgen.addcase("1")
     gen = str(sub.root)
     print gen
-    expected = ("SELECT CASE ( my_var )\n"
-                "CASE ( 1 )\n"
+    expected = ("      SELECT CASE ( my_var )\n"
```
Comment (rupertford) [inline src/tests/f2pygen_test.py]: change needed here as new version of parser adds in spaces appropriately for select.

**Comment by arporter** [comment]
All requested changes made. Ready for second review.

**Pull request merged**